### ISSUE TITLE
"Do not track" setting for contacts

### ISSUE THREAD

**Issue by dsp76** [mautic/mautic#2911]
What type of report is this:

| Q  | A
| ---| ---
| Bug report? | 
| Feature request? | Y
| Enhancement? | 

## Description:
Not just for receiving newsletters - but also for tracking user behaviour on a website, visitors need to opt in.  (at least in EU/Germany). As contacts may also be imported from other sources (e.g. leads from a trade show), there might be a state, where a contact is added to Mautic, but never having opted into tracking before. 

It would be good to have a "do not track" setting as well. "do not contact" is a different thing! This would allow Mautic users to still add such contact to a segment and send email and eventually get his opt-in to newsletter and website tracking through a form at a later time.

Initially discussed here: https://www.mautic.org/community/index.php/4047-feature-request-do-not-track

**Comment by npracht** [comment]
related to https://github.com/mautic/mautic/issues/3133

**Comment by npracht** [comment]
What type of report is this:

| Q  | A
| ---| ---
| Bug report? | 
| Feature request? | x
| Enhancement? | 

## Description:
In some countries and especially in Europe the law has been updated and clients (visitors of a website) must have the ability to disable cookie tracking.
![10438939](https://cloud.githubusercontent.com/assets/14075239/21449987/11f3d4f4-c8f3-11e6-80ce-96b024514dc5.jpg)

Consequently, it would be great to have a feature that help to disable cookie tracking by the visitor via a click on a button, something like `{disable_cookie}` and `{enable_cookie}`.

**Comment by dsp76** [comment]
Why is this closed @dbhurley ? Starting spring time in 2018 this missing feature might even block usage of Mautic in the EU.

**Comment by tritum** [comment]
If we would provide a PR would this be merged? Or can this be achieved by creating a plugin?

**Comment by PeterTL** [comment]
I think this CR is not constructive. In order to evaluate a "do not track" lead information, the lead in fact hast to be tracked, sent to the database and evaluated there. What I want to say: it is too late at this point.

I think the GDPR requires browser-level tracking code opt-in capabilities. This would mean: tracking script is disabled and opt-in message is displayed, if the user opts in to the tracking, a cookie with this information is stored to suppress the message in the future and the tracking script is loaded.

There a WordPress plugins which do exactly this for Google Analytics tracking. A similar functionality for Mautic would be perfect. But: it is something that will have to happen on script level, not directly within the Mautic application.

**Comment by dsp76** [comment]
@PeterTL okay, an opt-in solution within the website would help to prevent tracking. This could be done e.g. with Borlabs WordPress Cookie Plugin or similar solutions. 

As far as I know, opt-in is not necessary yet, so in a opt-out scenario it would not work.

It would be helpful to not record any website activity for a contact who doesn't want to be tracked. Of course - technically Mautic would receive the information related to a given contact, but you could decide not save it for a contact in their mautic contact history, when contact is set to "do not track". You may then use Mautic as your advanced newsletter system and for web analytics for contacts who didn't opt-out.

tl;dr You can't opt-out a single user from Mautic tracking and still send them newsletters with the same instance.

**Comment by ekkeguembel** [comment]
@dsp76 For contacts that are or can be personalised (be it through email or IP or other), opt-in is absolutely required already (by GDPR, not E-Privacy.)

So here's the scenario that would work, imho:
a) IP anonymization (#3445) is a must
b) Once you collect personal data, you also ask for tracking consent (part of the same form)
c) If consent is NOT given we need two things:
* purge existing tracking data from contact (urls visited, actions triggered, segment memberships, ...)
* prevent new data from being added

Please note that data for which you DO have legal reasons (e.g. email and name in a newsletter signup) can of course be stored. Just not any additional stuff.

**Comment by dsp76** [comment]
@ekkeguembel Your last sentence confirms the requirement. I can't differenciate between having a single contact just for newsletter subscription, but do not track him either. As soon as you have a contact in mautic, it will also track clicked links coming from the email. 

However, it would need to have some additional thoughts, as how to handle unsubscription, if a clicked link is not tracked...

So probably as of today, we only have the choice to have contacts in our newsletter system, who also agree to tracking.

**Comment by ekkeguembel** [comment]
@dsp76
> So probably as of today, we only have the choice to have contacts in our newsletter system, who also agree to tracking.

100% agreed! And that's exactly my point - because by this we are massively crippling the success of any usage (or are forced into illegality).

In other words, this technical limitation needs to be overcome quickly.

P.S. True, unsubscribe needs to be taken care of, too. Good point.

**Comment by rybskiryba** [comment]
Hi. Have you heard some update about this issue? I am super interested in the feature to disable cookies.

**Comment by npracht** [comment]
Hi guys.
This open source solution is integrated to mautic. https://opt-out.ferank.eu/en/
You can also use a SaaS version of them if you like.

**Comment by dinaras** [comment]
Hello, i have been reading this comment and would like to add a small detail. If you read closely the law in comment from [npracht](https://github.com/npracht) you will notice that it is talking about sending data to third party, ie mostly talking about google analytics. I believe that cookie consent is not needed for mautic if you host it yourself and not pass data to a third party.

**Comment by dsp76** [comment]
Not true according to my data privacy officer. The individual tracking is not necessary for your website to work. Thus the visitor must have a choice as soon its personal. By placing a cookie it gets personal and linked to the profile you record with Mautic.

We lately had a discussion about Matomo, which would also be self hosted. It depends, if you track with a cookie, or cookieless. This is a great article explaining it: [https://matomo.org/cookie-consent-banners/](https://matomo.org/cookie-consent-banners/).

We ended up by tracking cookieless without consent, only after consent we use cookie based tracking to improve data quality. However, while for web analytics this is a useful approach, its not adding much value for Mautic, as its all about profiles.
